main_test.py

```diff
@@ -5,58 +5,226 @@
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "F" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "G" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "H" }
+            ]
+        ],
+        "sid": 106,
+        "nickname": None,
+        "firstname": "Jonathan",
+        "lastname": "Briggs",
+        "gradyear": None,
+        "username": "jbriggs",
+        "advisor": None,
+        "grade": None
+    },
+}
+
+
+TEST_MASTER_SCHEDULE = [{
+    "2020-09-09": "Remote A-D_Rem",
+}]
+
+
+TEST_TEACHER = "jbriggs"
+TEST_STUDENT = "aaardvark"
+TEST_STUDENT_NO_PIC = "bbison"
+
+
+class FakeKey:
+    def __init__(self, name):
+        self.name = name
+
+
+class FakeEntity:
+    def __init__(self, key):
+        self.key = key
+    def get(self, prop):
+        return self.key.name != TEST_STUDENT_NO_PIC
+    def items(self):
+        return { x: self.get(x) for x in ['share_photo', 'share_schedule']}
+
+
+class FakeDatastore:
+    def key(self, a, b):
+        return FakeKey(b)
+    def get(self, key):
+        return FakeEntity(key)
+    def get_multi(self, keys):
+        return [FakeEntity(key) for key in keys]
+
+
+TEST_CONFIG = {
+    'TESTING': True,
+    'SECRET_KEY': bytearray('test-key', 'ascii'),
+    'TOKEN': 'test-token',
+    'SCHEDULES': TEST_SCHEDULES,
+    'MASTER_SCHEDULE': TEST_MASTER_SCHEDULE,
+    'DATASTORE': FakeDatastore()
+}
 
-''' Note - these tests run entirely on PRODUCTION data, so
-service account credentials are needed for them to pass.
-This stems from my belief that tests not using fake data
-are less scientifically valid (i.e. drug testing on humans
-is more valid than testing on mice) and from a desire to
-not have production and test versions of each dataset'''
 
 API_ENDPOINTS = [
-    "/student/guberti",
+    "/student/aaardvark",
     "/class/a",
     "/period/a",
-    "/search/guberti"
+    "/search/aaardvark"
 ]
 
-
-def download_photo(url):
-    r = requests.get(url, stream=True)
-    return Image.open(BytesIO(r.content))
-
 class NoAuthTests(unittest.TestCase):
     @classmethod
     def setUpClass(self):
-        app.config['TESTING'] = True
+        init_app(TEST_CONFIG)
         self.client = app.test_client()
 
+    # Test that when not logged in, we're given a sign-in page.
     def test_main_login_response(self):
         response = self.client.get('/')
         self.assertEqual(response.status_code, 200)
         # TODO kinda fragile
         self.assertIn("Sign in to EPSchedule", str(response.data))
 
+    # Test that when not logged in, all API endpoints fail with a 403.
     def test_api_unauthorized(self):
         for endpoint in API_ENDPOINTS:

```

Comment:
We should use `self.subTest` here so that if this test fails, it produces a useful error message.